sphinx/writers/latex.py

```diff
@@ -365,13 +388,22 @@ def get_colspec(self):
             return '{|' + ('l|' * self.colcount) + '}\n'
 
     def add_cell(self, height, width):
+        """Adds a new cell to a table.
+
+        It will be located at current position: (``self.row``, ``self.col``).
+        """
         self.cell_id += 1
         for col in range(width):
             for row in range(height):
                 assert self.cells[(self.row + row, self.col + col)] == 0
                 self.cells[(self.row + row, self.col + col)] = self.cell_id
```

Comment:
I am a bit worried that if we add at position (row=5, col=5) a cell of height and width 2, then if we add at (row=5, col=3) a cell of height 3 and width 6 it will lead to the new cell_id being assigned to non rectangular area. If I get it wright this means we must use add_cell in specific order ?